java/source/io.rst

```diff
@@ -443,3 +443,108 @@ Reading Parquet File
+        myAppUseCountryDictionary.set(3, "Malta".getBytes(StandardCharsets.UTF_8));
+        myAppUseCountryDictionary.set(4, "Uganda".getBytes(StandardCharsets.UTF_8));
+        myAppUseCountryDictionary.setValueCount(5);
+        System.out.println("Data to retain: " + myAppUseCountryDictionary);
+
+        File file = new File("randon_access_to_file_with_dictionary.arrow");
+        DictionaryProvider.MapDictionaryProvider provider = new DictionaryProvider.MapDictionaryProvider();
+        provider.put(myCountryDictionary);
+        try (FieldVector myAppUseCountryDictionaryEncoded = (FieldVector) DictionaryEncoder
+                .encode(myAppUseCountryDictionary, myCountryDictionary);
+             VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.of(myAppUseCountryDictionaryEncoded);
+             FileOutputStream fileOutputStream = new FileOutputStream(file);
+             ArrowFileWriter writer = new ArrowFileWriter(vectorSchemaRoot, provider, fileOutputStream.getChannel())
+        ) {
+            System.out.println("Data to retain through Dictionary: " +myAppUseCountryDictionaryEncoded);
+            writer.start();
+            writer.writeBatch();
+            writer.end();
+            System.out.println("Record batches written: " + writer.getRecordBlocks().size() + ". Number of rows written: " + vectorSchemaRoot.getRowCount());
+            try(
+                    BufferAllocator rootAllocator = new RootAllocator();
+                    FileInputStream fileInputStream = new FileInputStream(file);
+                    ArrowFileReader reader = new ArrowFileReader(fileInputStream.getChannel(), rootAllocator)
+            ){
+                for (ArrowBlock arrowBlock : reader.getRecordBlocks()) {
+                    reader.loadRecordBatch(arrowBlock);
+                    FieldVector myAppUseCountryDictionaryEncodedRead = reader.getVectorSchemaRoot().getVector("app-use-country-dict");
+                    Dictionary myAppUseCountryDictionaryRead = reader.getDictionaryVectors().get(123L);
+                    System.out.println("Data to retain recovered: " + myAppUseCountryDictionaryEncodedRead);
+                    System.out.println("Dictionary recovered: " + myAppUseCountryDictionaryRead);
+                    try (ValueVector readVector = DictionaryEncoder.decode(myAppUseCountryDictionaryEncodedRead, myAppUseCountryDictionaryRead)) {
+                        System.out.println("Data to retain recovered decoded: " + readVector);
+                    }
+                }
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+.. testoutput::
+
+    Dictionary used: Dictionary DictionaryEncoding[id=123,ordered=false,indexType=Int(32, true)] [Andorra, Cuba, Grecia, Guinea, Islandia, Malta, Tailandia, Uganda, Yemen, Zambia]
+    Data to retain: [Andorra, Guinea, Islandia, Malta, Uganda]
```

Comment:
In all of these examples I would call this "Unencoded data" and "Dictionary-encoded data"